### DISCUSSION TITLE
.Net MAUI Blazor debugging

### DISCUSSION THREAD

**Discussion by mabanza** [dotnet/maui#3287] [Q&A]
Hi there. 
In my .Net MAUI Blazor application, how can I debug, say my CSS or JavaScript in the browser? In regular Edge or Chrome browser I can hit F12 and see it under the hood. Is it possible to do the same in .Net MAUI Blazor app? Thanks.

**Comment by julianadormon** [comment]
https://www.reddit.com/r/Blazor/comments/ox1q2m/how_to_use_browser_dev_tools_to_debug_net_maui/

**Reply by tharlab** [reply]
not windows mechine/wbview emulator, sometime we need real debug in real browser and test it on different browser like chome, firefox etc

**Comment by tharlab** [comment]
yes we need this, sometime we need real debug in real browser and test it on different browser like chome, firefox etc
not windows mechine/wbview emulator, also hot reload doesn't work in windows mechine/wbview emulator,

**Comment by mabanza** [comment]
May be I was not accurate enough putting down my question. Yes, we all know how F12 works, but I was asking about .NET MAUI Blazor - the thing we can use for mobile development. While developing for mobile, where is the F12 button on my, say, Samsung A5? Thanks

**Comment by Martinn2** [comment]
In Ionic framework I use chrome://inspect/#devices to debug native app in chrome browser. I am not sure if it works for MAUI.
